### PR TITLE
Disable storage

### PR DESCRIPTION
**Pull request by brightbyte**
Allow storage of parsoid output to be disabled based on the domain parameter

### CODE REVIEW

File: config.fullstack.test.yaml [hunk 1]
```diff
@@ -7,8 +7,9 @@ default_project: &default_project
               - path: projects/v1/default.wmf.yaml
                 options: &default_options
                   parsoid:
-                    host: http://parsoid-external-ci-access.beta.wmflabs.org/w/rest.php
-                    grace_ttl: 1000000
+                    host: https://parsoid-external-ci-access.beta.wmflabs.org/w/rest.php
+                    # disabled_storage: true
+                    grace_ttl: 1000007
```
Comment (brightbyte) [inline config.fullstack.test.yaml:12]: These changes should probably not be in this patch.
Though I can't test without https...

File: projects/sys/default.wmf.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ paths:
       - path: sys/parsoid.js
         options:
           host: '{{options.parsoid.host}}'
+          disabled_storage: '{{options.parsoid.disabled_storage}}'
```
Comment (brightbyte) [inline projects/sys/default.wmf.yaml:17]: is this actually needed? not sure...

File: test/features/pagecontent/language_variants.js [hunk 3]
```diff
@@ -158,7 +158,7 @@ describe('Language variants', function() {
             assert.deepEqual(res.headers['content-language'], 'de');
             assert.deepEqual(res.headers['x-restbase-sunset'] || null, 'true');
             assert.checkString(res.headers.etag, /^"\d+\/[a-f0-9-]+"$/);
-            assert.deepEqual('Das ist eine testseite', res.body.extract);
+            assert.deepEqual('Das ist eine testseite.', res.body.extract);
```
Comment (brightbyte) [inline test/features/pagecontent/language_variants.js:161]: The content of the page on beta was actually changed! Took me forever to figure that out...

https://de.wikipedia.beta.wmflabs.org/w/index.php?title=RESTBase_Testing_Page&diff=prev&oldid=30418

File: test/features/pagecontent/rerendering.js
Comment (brightbyte) [inline test/features/pagecontent/rerendering.js:1]: should not be in this patch

File: test/features/pagecontent/pagecontent.js
Comment (brightbyte) [inline test/features/pagecontent/pagecontent.js:1]: should not be in this patch

File: sys/parsoid.js [hunk 4]
```diff
@@ -519,47 +544,63 @@ class ParsoidService {
...
-                        status: 500,
-                        body: {
-                            title: 'no_etag',
-                            description: 'No ETag has been provided in the response'
-                        }
-                    });
+                    if (disabledStorage) {
+                        // Generate an ETag, for consistency
+                        const uuid = uuidv1();
+                        res.headers.etag = mwUtil.makeETag(res.headers['content-revision-id'] || '0', uuid);
```
Comment (johngian) [inline sys/parsoid.js:588]: I think here we are missing some DOM manipulation that't happening on `generateAndSave` to pass the tid to html meta.

Comment (johngian) [reply sys/parsoid.js:588]: https://github.com/wikimedia/restbase/blob/6a8e93390e59552e01d296e0563c9ef3ae1867bb/sys/parsoid.js#L444

**Review by johngian — CHANGES_REQUESTED** [review]

**Comment by johngian** [comment]
Do we have a phab ticket for reference?

**Comment by brightbyte** [comment]
> Do we have a phab ticket for reference?

https://phabricator.wikimedia.org/T344945

**Comment by brightbyte** [comment]
> It might also be useful to add a header for the fact that parsoid storage is disabled on responses and add a test to assert that for a test wiki from the local env setup.

Something like `x-restbase-cache-disabled`?

File: sys/parsoid.js [hunk 5]
```diff
@@ -391,10 +403,23 @@ class ParsoidService {
...
-            `page/pagebundle/${encodeURIComponent(rp.title)}/${rp.revision}`
-        ));
+            path
+        );
+
+        return hyper.get(parsoidReq)
+          .then((resp) => {
+              return resp;
+          }, (error) => {
+              throw error;
```
[the remark this answers is not in the document]
Comment (brightbyte) [reply sys/parsoid.js:421]: I don't remember exactly, but I often do this kind of thing when I want to be able to set a break point.

**Comment by johngian** [comment]
> > It might also be useful to add a header for the fact that parsoid storage is disabled on responses and add a test to assert that for a test wiki from the local env setup.
> 
> Something like `x-restbase-cache-disabled`?

Yeah, so we know what kind of requests we are getting (in case we need to debug from the client side) and also to be able to test this.

**Comment by brightbyte** [comment]
> > Something like `x-restbase-cache-disabled`?
> 
> Yeah, so we know what kind of requests we are getting (in case we need to debug from the client side) and also to be able to test this.

I implemented it in the latest version.

**Comment by brightbyte** [comment]
For the record, I don't understand why the "should render & re-render independent revisions" test is now failing with storage enabled. The behavior for that case shouldn't have changed at all. Did I actually break something, or is it an unrelated issue?...

**Review by johngian — APPROVED** [review]

**Pull request merged**